lib/syntax.ml

```diff
@@ -255,15 +256,13 @@ and assign_types expr =
         | Group typ, _ -> typ, types
         | Agg, _ -> fail "cannot use this grouping function with %d parameters" (List.length types)
         | F (_, args), _ when List.length args <> List.length types -> fail "wrong number of arguments : %s" (show_func ())
-        | Coalesce (ret, each_arg) , _ -> 
-          let args = types_to_arg each_arg in
+        | F (ret, args), _ when Stdlib.(name = Name "coalesce") -> 
```

Comment:
i don't like this approach, precisely because of this ^
it will become unmanageable very fast
everything that is needed for type/nullability inference should be in the type, not conditioned on function names at random places

Reply:
Fixed, got rid of binding to names, and having names at all